src/include/distributed/tuple_destination.h

```diff
@@ -0,0 +1,47 @@
+/*-------------------------------------------------------------------------
+ *
+ * tuple_destination.h
+ *	  Tuple destination generic struct.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef TUPLE_DESTINATION_H
+#define TUPLE_DESTINATION_H
+
+#include "access/tupdesc.h"
+#include "distributed/multi_physical_planner.h"
+#include "tcop/dest.h"
+#include "utils/tuplestore.h"
+
+typedef struct TupleDestination TupleDestination;
```

Comment:
This PR created 2 set of warnings in my setup. Both are related to the redefinition of `TupleDestination`

1 of 2:

```
In file included from commands/multi_copy.c:74:
In file included from /Users/hanefi/code/citus/citus/src/include/distributed/local_executor.h:15:
/Users/hanefi/code/citus/citus/src/include/distributed/tuple_destination.h:39:3: warning: redefinition of typedef 'TupleDestination' is a C11 feature [-Wtypedef-redefinition]
} TupleDestination;
  ^
/Users/hanefi/code/citus/citus/src/include/distributed/tuple_destination.h:18:33: note: previous definition is here
typedef struct TupleDestination TupleDestination;
                                ^
```